torch/serialization.py

```diff
@@ -1320,7 +1320,11 @@ def restore_location(storage, location):
 
 class StorageType:
     def __init__(self, name):
-        self.dtype = _get_dtype_from_pickle_storage_type(name)
+        self._dtype = _get_dtype_from_pickle_storage_type(name)
+
+    @property
+    def dtype(self):
```

Comment:
Why is this different?

Reply:
Because I don't want to replicate weird logic between legacy vs non-legacy deserialization or make any tweaks to this logic: 
https://github.com/pytorch/pytorch/blob/79e3833703c421cb9cb9ec2c45751a09abbe6e9b/torch/serialization.py#L1135

Reply:
Sounds fair!